src/api/java/com/minecolonies/api/research/IResearchManager.java

```diff
@@ -36,4 +37,11 @@
      * @return the ResearchEffects object.
      */
     IResearchEffectManager getResearchEffects();
+
+    /**
+     * Checks if any autostart research has its prerequisites filled,
+     * and if so, prompts the player for resources or begins research if no resources required.
+     * @param colony       The colony where the research is being checked.
+     */
+    void checkAutoStartResearch(IColony colony);
```

Comment:
ResearchManagers exist within the Colony, but they don't have the colony as an internally-accessible property the way that RequestManagers do.  I can change that if it'd be considered a cleaner option, but I wasn't sure if that would be preferable given the colony/colonyview split.

Reply:
If the manager is inheritable linked to the colony then it should have the colony internally.

If there is a need for you to have a view then your research manager needs a view class of it's own

Reply:
I can make that change if desired, but ResearchManager may be linked to Colony or a ColonyView depending on where it's created, and there are a lot of ColonyView properties that will return nulls where Colony would not, many of them places it would be really bulky to do the necessary null or instanceof checks.

That's not a problem for checkAutoStartResearch (or startCostlessResearch), which neither request or would need something that fails that problem, and they're always called on the server-side anyway.  But it seems like setting up a landmine for other coders.